### PR TITLE
Fix: this.state.chans and channel.users are Map type, not an Object type

### PR DESCRIPTION
**Pull request by iwalkalone69**
- Object.entries(this.state.chans) is not working as it is a Map, it returns an empty array.
- message.nick in killChannel.users is not working as it is a Map too. Repacing with Map.prototype.has().
- Replacing delete killChannel.users[message.nick] with Map.prototype.delete().
- Doing the same with quitChannel.

### CODE REVIEW

**Comment by progval** [comment]
So that's why appservice-irc stopped bridging quits months ago! Thanks @iwalkalone69!

**Comment by progval** [comment]
Did you check `onNick` doesn't need it too?

**Comment by iwalkalone69** [comment]
> Did you check `onNick` doesn't need it too?

I have committed another change that fixes this. It uses Map.prototype.forEach() to iterate channels.

**Comment by Half-Shot** [comment]
Incidentally I think this sort of thing really needs a linting rule, but alas there isn't one yet https://github.com/typescript-eslint/typescript-eslint/issues/6807

File: src/irc.ts [hunk 1]
```diff
@@ -618,10 +618,9 @@ export class Client extends (EventEmitter as unknown as new () => TypedEmitter<C
         const channelsForNick: string[] = [];
 
         // finding what channels a user is in
-        Object.entries(this.state.chans).forEach(([channame, nickChannel]) => {
-            const chanUser = message.nick && nickChannel.users.get(message.nick);
-            if (message.nick && chanUser) {
-                nickChannel.users.set(message.args[0], chanUser);
+        this.state.chans.forEach((nickChannel, channame) => {
+            if (message.nick && nickChannel.users.has(message.nick)) {
+                nickChannel.users.set(message.args[0], nickChannel.users.get(message.nick));
```
Comment (Half-Shot) [inline src/irc.ts:623]: Seems like there is a typing fail here?

Comment (iwalkalone69) [reply src/irc.ts]: Yes, sorry. It is fixed now.

**Review by Half-Shot — APPROVED** [review]
Thanks for the bugfix :)

**Pull request merged**